lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -66,6 +67,10 @@ def test_matrix(self):
         self.assertEqual(type(cube.data), np.ndarray)
         self.assertArrayEqual(cube.data, data)
 
+    def test_default_share_data(self):
+        cube = Cube(np.arange(1))
+        self.assertFalse(cube.share_data)
```

Comment:
`cube.share_data` is part of the Iris public API
https://github.com/SciTools/iris/blob/v1.13.0/lib/iris/cube.py#L780

it was proposed by a contributor and reviewed and merged by me; it was included in the 1.13 release.
This followed a long and detailed set of feature requirements, that was drastically paired down to try and minimise impact for version 2

I do not think it is appropriate to remove the part of the API in the next version of Iris. There has been no deprecation warning or any thought on alternative API options to retain the required functionality.